include/rmm/mr/device/per_device_resource.hpp

```diff
@@ -121,6 +129,25 @@ RMM_EXPORT inline auto& get_map()
   return device_id_to_resource;
 }
 
+/**
+ * @briefreturn{Reference to the lock}
+ */
+inline std::mutex& ref_map_lock()
+{
+  static std::mutex ref_map_lock;
```

Comment:
*question*: Here, we have a global static reference, but this function has hidden visibility. I _think_ that might mean that (as per #826), if one links RMM both via C++ and python, then there are two versions of the lock object floating around?

So (plausibly) all these `inline` functions that return references to global statics should be `RMM_EXPORT`.

Reply:
Yeah, I did:
```patch
diff --git a/include/rmm/mr/device/per_device_resource.hpp b/include/rmm/mr/device/per_device_resource.hpp
index a56a784a..d5afafe4 100644
--- a/include/rmm/mr/device/per_device_resource.hpp
+++ b/include/rmm/mr/device/per_device_resource.hpp
@@ -108,6 +108,7 @@ inline device_memory_resource* initial_resource()
 inline std::mutex& map_lock()
 {
   static std::mutex map_lock;
+  std::cout << "Map lock " << &map_lock << std::endl; 
   return map_lock;
 }
```

And running a simple cudf example:
```python
In [1]: import cudf

In [2]: df = cudf.DataFrame({"id": [1, 2, 3], "col0": [0, 1, 2], "col1": [1, 2, 3], "col2": [2, 3, 4]})
Map lock 0x7355ec744e20
Map lock 0x7355bc40c400
Map lock 0x7355ec744e20
Map lock 0x7355bc40c400
Map lock 0x7355ec744e20
Map lock 0x7355bc40c400
Map lock 0x7355ec744e20
Map lock 0x7355bc40c400
```

So there are two locks, I will open a separate PR to fix these issues.

Reply:
Good catch!

Reply:
Done.